talk/owt/sdk/p2p/p2pclient.cc

```diff
@@ -293,24 +293,40 @@ void P2PClient::OnSignalingMessage(const std::string& message,
         rtc::GetIntFromJsonObject(stop_info, "code", &code);
         rtc::GetStringFromJsonObject(stop_info, "message", &error);
 
+        if (code == kWebrtcIceGatheringPolicyUnsupported) {
+          auto pcc = that->GetPeerConnectionChannel(remote_id);
+          std::shared_ptr<LocalStream> stream = pcc->GetLatestLocalStream();
+          std::function<void()> success_callback =
+              pcc->GetLatestPublishSuccessCallback();
+          std::function<void(std::unique_ptr<Exception>)> failure_callback =
+              pcc->GetLatestPublishFailureCallback();
+          pcc->SetAbandoned();
+          that->pc_channels_.erase(remote_id);
```

Comment:
Could you clarify why locking is not neccessary for this?

Reply:
Fixed. Locking is required here.